baseplate/thrift_pool.py

```diff
@@ -37,13 +37,22 @@
 logger = logging.getLogger(__name__)
 
 
-def _make_protocol(endpoint):
+def _make_protocol(endpoint, transport_wrapper=None):
+    """
+    The transport_wrapper parameter passed in here should be an instance of a
+    factory that allows us to wrap the TSocket implementation in a
+    BufferedTransport or FramedTransport if necessary.
+    """
     if endpoint.family == socket.AF_INET:
         trans = TSocket.TSocket(*endpoint.address)
     elif endpoint.family == socket.AF_UNIX:
         trans = TSocket.TSocket(unix_socket=endpoint.address)
     else:
         raise Exception("unsupported endpoint family %r" % endpoint.family)
+
+    if transport_wrapper is not None:
```

Comment:
I'm not sure it's actually safe to put TFramedTransport beneath THeaderProtocol in terms of actually getting bytes on the wire that make sense to any server. Instead, I think we should move the instantiation of the `THeaderProtocol` into the default `transport_wrapper` (which would be better called `protocol_factory`) and let your application override that to just `TFramedTransport` being wrapped by `TBinaryProtocol`. 


```python
def __init__(self, endpoint, size..., transport_factory=THeaderProtocol.THeaderProtocol):
    ...
    self.protocol_factory = protocol_factory

def _acquire(self):
    ...
    trans = _make_transport(self.endpoint)
    prot = self.protocol_factory(transport)
    ...

def _make_transport(endpoint):
    if endpoint.family ...
    ...
    return trans
```

```python
def framed_protocol_factory(trans):
    trans = TFramedTransport(trans)
    return TBinaryProtocol(trans)

my_framed_pool = ThriftConnectionPool(endpoint, protocol_factory=framed_protocol_factory)
```